src/main/java/io/netty/incubator/codec/quic/QuicChannel.java

```diff
@@ -148,6 +149,13 @@ default ChannelPromise voidPromise() {
     @Override
     QuicChannelConfig config();
 
+    /**
+     * Returns the used {@link SSLEngine} or {@code null} if none is used (yet).
+     *
+     * @return the engine.
+     */
+    SSLEngine sslEngine();
```

Comment:
Since the potential `null` here is just because of timing, should this return a `Future` instead of a nullable return value?

Reply:
As we may allow to use QUIC without TLS at some point I don't think so ... I did consider an Optional tho

Reply:
ok in that case should we eventually be moving this to the base `Channel` API? I am wondering why is this special for QUIC